### ISSUE TITLE
Suggestion: option to disable IDE (or sdcard) image hash at startup. 10 min to do it in my PC

### ISSUE THREAD

**Issue by novachip1** [openMSX/openMSX#1119]
Before starting emulation, my pc takes about 10 minutes to hash an image of the 4gb ide (sunrise) extension. So far as I know, to make reverse / rewind possible. This could have an option to be disabled so it does not take so long. 

Thank you
-Leo from São Paulo / Brazil

**Comment by MBilderbeek** [comment]
We can do several things:
1) User disables reverse when using large harddisk images manually. That's a workaround
2) We create a setting which tells from which media size, reverse should be automatically disabled if such media is being used. The problem is that currently reverse is a global feature. So if you are running multiple machines concurrently (yes this is possible in openMSX!) all machines are affected. But perhaps we can find something to cope with this.
3) we stop hashing harddisk images (which is basically what you are suggesting in the title of this ticket). The reason we are hashing them is to know whether they changed, which is very important for reverse. E.g. if you replay a replay file with a harddisk image which has been changed in the mean time the replayed writes to the harddisk image can severely corrupt the harddisk image. At the same time, replaying a replay with a changed harddisk will probably not sync, as the media is different then when the recording (replay file) was made. So, this is not something you want!
4) we implement some kind of fancy copy-on-write mechanism to avoid corruption of harddisk images when replaying. This is probably the way to go, but far from trivial, so it's not easily implemented.

In other words: this is a tricky one.

**Comment by Vampier** [comment]
I create a 16mb hdd image for smaller games I run on HDD and I want to use reverse for. So it could be a combination of things. I would not hash anything over 512mb since it's just a waste of time. Even if you don't change anything on the disk the sha1 checksum might still change (unless you make it read only)

on a Raspberry Pi it took even longer and I disabled reverse on openmsx boot.

**Comment by novachip1** [comment]
Hello! As for disabling "reverse" for all instances of machines, I
believe this is not a problem (at least for me). See what can be done
without causing a lot of headaches, from a parameter at startup, to a
checkbox in Catapult to avoid the hash, would be of great value! The
suggestion is mine and the creativity is yours!

Thank you!

**Comment by MBilderbeek** [comment]
@novachip1 A question: is that 4GB on the local harddrive of the PC?
Is it still 10 minutes with the current code? There have been some performance improvements.

**Comment by MBilderbeek** [comment]
@novachip1 Can you respond, please?

**Comment by MBilderbeek** [comment]
Closing as we got no response from @novachip1 . Please reopen if you can help us further with this.
